pymc/distributions/distribution.py

```diff
@@ -615,3 +618,18 @@ def func(*args, **kwargs):
         raise NotImplementedError(message)
 
     return func
+
+
+def default_get_moment(rv, size, *rv_inputs, rv_name=None, has_fallback=False):
+    moment = at.zeros(size, dtype=rv.dtype)
+    if moment.ndim == rv.ndim:
```

Comment:
Mulling it over, I am not sure we can safely assume we got the right `size` by simply checking the `ndim` matches. 

The shape may be determined by the `rv_inputs` and not the size:

```python
pm.DensityDist('x', np.arange(5), logp=lambda value, rv_input1: ..., size=None)
```

We can perhaps use the `rv._infer_shape` method, which seems to do just that: 
https://github.com/aesara-devs/aesara/blob/d9fd640e04164759ac752333b94b294a0a5c5117/aesara/tensor/random/op.py#L162

I would still be weary of relying on that for multivariate variables because that often requires overriding for correct multivariate RandomVariable shape inference

Reply:
I guess this depends on whether something like `pm.Normal(..., mu=np.arange(3), size=())` will work, and broadcast into something that has shape `(3,)`. From what I understand, numpy and scipy forbid this, and always force the `size` to match the shape induced by the distribution parameters.

```python
>>> np.random.default_rng().normal(loc=np.arange(3), scale=1, size=())

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)

----> 1 np.random.default_rng().normal(loc=np.arange(3), scale=1, size=())

_generator.pyx in numpy.random._generator.Generator.normal()

_common.pyx in numpy.random._common.cont()

_common.pyx in numpy.random._common.cont_broadcast_2()

_common.pyx in numpy.random._common.validate_output_shape()

ValueError: Output size () is not compatible with broadcast dimensions of inputs (3,).
```

Reply:
Unfortunately Aesara behaves differently than Numpy/Scipy, since the later are quite restrictive: https://github.com/aesara-devs/aesara/discussions/412